docs/source/app-dev/app-arch.rst

```diff
@@ -246,3 +235,79 @@ There are some other identifiers that are determined by your client code. These
 .. |image0| image:: images/BotFlow.png
    :width: 6.5in
    :height: 3.69444in
+
+Testing
+=======
+
+Testing is fundamental to ensure correctness and favor maintainability.
+
+Testing is usually divided in different categories according to its scope and aim:
+
+- unit testing verifies single properties of individual components
+- integration testing verifies that an aggregation of components behaves as expected
+- acceptance testing checks that the overall behavior of a whole system satisfies certain criteria
+
+Both tests in the small scale (unit testing) and large (acceptance testing) tend to be specific to the given component or system under test.
+
+This chapter focuses on providing portable approaches and techniques to perform integration testing between your components and an actual running ledger.
+
+Test the business logic with a ledger
```

Comment:
I don't think you need a new paragraph for each sentence in this section. I'd structure it in three paragraphs:

1. DAML Contracts + DAML Ledger Model encapsulate the bulk of your business logic
2. Mocking the Ledger API pretty much requires you to reimplement a full DAML Ledger so don't.
3. The Sandbox is your friend. 

In 3. you could explain options for getting the ledger into a desired state:
  - Write a scenario
  - Write a special ledger initialisation workflow and automation
  - Use a Postgres sandbox with initial state taken from a backed up database.

Reply:
I would strongly discourage writing a scenario because of the [limitations of using scenarios with the suggested approach](https://github.com/digital-asset/daml/pull/2505/files/fe13e50529b6718a11b35d913703b2df7fb60959#r313346380).

Reply:
Where does the Postgres backup idea come from? It's creative but I would not recommend it unless there are some established good practices around it. If there are, please share them with me so that I can add them.

Reply:
Just from general brainstorming around testing. Defining, and loading the desired ledger state is one of the more cumbersome parts currently so it's on my radar as something to offer solutions for. 

Admittedly, I've never actually tried it.